src/openrct2/scripting/ScNetwork.hpp

```diff
@@ -447,6 +449,38 @@ namespace OpenRCT2::Scripting
 #    endif
         }
 
+#    ifndef DISABLE_NETWORK
+        std::shared_ptr<ScSocketServer> createServer()
+        {
+            auto& scriptEngine = GetContext()->GetScriptEngine();
+            auto plugin = scriptEngine.GetExecInfo().GetCurrentPlugin();
+            auto socket = std::make_shared<ScSocketServer>(plugin);
+            scriptEngine.AddSocket(socket);
+            return socket;
+        }
+#    else
+        void createServer()
+        {
+            duk_error(_context, DUK_ERR_ERROR, "Networking has been disabled.");
+        }
+#    endif
+
+#    ifndef DISABLE_NETWORK
+        std::shared_ptr<ScSocket> createSocket()
```

Comment:
I think it would be better to call this client rather than socket. A socket can be both client and server.

Reply:
Well node.js has Socket and Server, hence my Socket and Server JS classes, so I was more or less trying to match that with these creation methods.